modules/source/source_test.py

```diff
@@ -29,89 +29,94 @@
 # NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
 # SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
+import queue
+import re
 import unittest
-import Queue
+
 import config
-import re
-import logging
-import source
+from . import source
+
 
 class InvalidChannelExceptionMock(Exception):
     pass
 
+
 class StateMock():
-    def __init__(self, cid = 0, session = 0, userid = -1):
+    def __init__(self, cid=0, session=0, userid=-1):
         self.channel = cid
         self.session = session
         self.userid = userid
 
+
 class ChannelStateMock():
     def __init__(self, cid, name, parent, groups, acls):
         self.id = cid
         self.name = name
         self.parent = parent
         self.groups = groups
         self.acls = acls
-        
+
+
 class ServerMock():
     def __init__(self, sid):
         self.sid = sid
         self._reset()
-    
+
     def id(self):
         return self.sid
-    
+
     def _lastChannelID(self):
         return self.uid
-        
+
     def addChannel(self, name, parent):
         self.uid += 1
-        assert(not self.uid in self.channels)
+        assert (not self.uid in self.channels)
         self.channels[self.uid] = ChannelStateMock(self.uid,
                                                    name,
                                                    parent,
                                                    {},
                                                    [])
         return self.uid
-    
+
     def addUserToGroup(self, cid, session, group):
         c = self._getChan(cid)
-        
+
         if session in c.groups:
             c.groups[session].add(group)
         else:
-            c.groups[session] = set([group])
-    
+            c.groups[session] = {group}
```

Comment:
This is confusing to me. It was a set, now it’s an object/dictionary?

Reply:
Actually, that is [one way to create a set](https://docs.python.org/3/tutorial/datastructures.html#sets). So no chnage here, just a more 'pythonic' syntax.

Reply:
`{}` dict, `{1}` set, `{'a':1}` dict

ok…